cdap-standalone/src/main/java/co/cask/cdap/TrackerAppCreationService.java

```diff
@@ -87,9 +93,15 @@ public Boolean call() throws Exception {
           }
           return artifacts != null && !artifacts.isEmpty();
         }
-      }, 5, TimeUnit.MINUTES, 5, TimeUnit.SECONDS, "Waiting for Tracker Artifact to become available.");
+      }, 5, TimeUnit.MINUTES, 2, TimeUnit.SECONDS, "Waiting for Tracker Artifact to become available.");
       // Find the latest tracker artifact based on the version
       List<ArtifactSummary> artifacts = new ArrayList<>(artifactRepository.getArtifacts(NamespaceId.SYSTEM, "tracker"));
```

Comment:
could this call be null `artifactRepository.getArtifacts(NamespaceId.SYSTEM, "tracker")` ? you were checking for null on L#94,  if it can be null, this will throw an exception.